dataplane/linux/vxlan_mgr.go

```diff
@@ -155,6 +172,13 @@ func (m *vxlanManager) OnUpdate(protoBufMsg interface{}) {
 			m.routesByDest[msg.Dst] = msg
 			m.routesDirty = true
 		}
+
+		// Process IPAM blocks that aren't associated to a single or /32 local workload
+		if msg.Type == proto.RouteType_LOCAL_WORKLOAD && msg.IpPoolType == proto.IPPoolType_VXLAN && !msg.LocalWorkload {
```

Comment:
Just thought of a corner case.  If you have a /32 block then it can oscillate between being LocalWorkload=true and LocalWorkload=false.

Fix: add an else block to this that does 
```
else if _, ok := m.ipamBlocks[msg.Dst]; ok {
 logrus.WithField("msg", msg).Info("VXLAN data plane IPAM block changed to something else")
  delete(m.ipamBlocks, msg.Dst)
￼  m.routesDirty = true
}
```